pkg/hns/endpoint_windows.go

```diff
@@ -40,6 +51,92 @@ func GetSandboxContainerID(containerID string, netNs string) string {
 	return containerID
 }
 
+func GenerateHnsEndpoint(epInfo *EndpointInfo, n *NetConf) *hcsshim.HNSEndpoint {
+	// run the IPAM plugin and get back the config to apply
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(epInfo.EndpointName)
+	if hnsEndpoint != nil && err != nil {
+		if hnsEndpoint.VirtualNetwork != epInfo.NetworkId {
+			_, err = hnsEndpoint.Delete()
+			if err != nil {
+				errors.Annotatef(err, "[win-cni] Failed to delete endpoint %v, err:%v", epInfo.EndpointName, err)
+			}
+		} else {
+			errors.Annotatef(err, "[win-cni] Found existing endpoint %v", epInfo.EndpointName)
+		}
+	} else {
+
+		hnsEndpoint = &hcsshim.HNSEndpoint{
+			Name:           epInfo.EndpointName,
+			VirtualNetwork: epInfo.NetworkId,
+			DNSServerList:  strings.Join(epInfo.Nameservers, ","),
+			DNSSuffix:      strings.Join(epInfo.DnsSearch, ","),
+			GatewayAddress: epInfo.Gateway.String(),
+			IPAddress:      epInfo.IpAddress,
+			Policies:       n.MarshalPolicies(),
+		}
+
+		errors.Annotatef(err, "Adding Hns Endpoint %v", hnsEndpoint)
+	}
+	return hnsEndpoint
+}
+
+func GenerateHcnEndpoint(epInfo *EndpointInfo, n *NetConf) *hcn.HostComputeEndpoint {
+	// run the IPAM plugin and get back the config to apply
+	hcnEndpoint, err := hcn.GetEndpointByName(epInfo.EndpointName)
+	if hcnEndpoint != nil && err != nil {
+		if hcnEndpoint.HostComputeNetwork != epInfo.NetworkId {
+			err = hcnEndpoint.Delete()
+			if err != nil {
+				errors.Annotatef(err, "[win-cni] Failed to delete endpoint %v, err:%v", epInfo.EndpointName, err)
+			}
+		} else {
+			errors.Annotatef(err, "[win-cni] Found existing endpoint %v", epInfo.EndpointName)
+		}
+	} else {
+
+		routes := []hcn.Route{
+			hcn.Route{
```

Comment:
Just a style thing but the `hcn.Route` on this line is no longer needed in Golang since the slice type is declared